src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp

```diff
@@ -185,6 +188,95 @@ namespace OpenMS
     return integral;
   }
 
+  PeakIntegrator::PeakShapeMetrics_ PeakIntegrator::calculatePeakShapeMetrics(
+    const MSChromatogram& chromatogram,
+    const double& left,
+    const double& right
+  ) const
+  {
+    // TODO remove the three following lines
+    // peak_height_ = 965356;
+    // peak_apex_rt_ = 2.7045;
+    // background_height_ = 723.5;
+    PeakShapeMetrics_ peakShapeMetrics;
+    peakShapeMetrics.points_across_baseline = 0;
+
+    const double start_intensity = std::max(chromatogram.RTBegin(left)->getIntensity() - background_height_, 0.0);
+    const double end_intensity = std::max((chromatogram.RTEnd(right)-1)->getIntensity() - background_height_, 0.0);
+    // TODO remove the following couts
+    std::cout << std::endl << "int left: " << chromatogram.RTBegin(left)->getIntensity() << std::endl;
+    std::cout << "int right: " << (chromatogram.RTEnd(right)-1)->getIntensity() << std::endl;
+    std::cout << "bg_height: " << background_height_ << std::endl;
+    std::cout << "start_intensity: " << start_intensity << std:: endl;
+    std::cout << "end_intensity: " << end_intensity << std::endl;
+
+    MSChromatogram::ConstIterator it_rtbegin_l = chromatogram.RTBegin(left);
```

Comment:
I am not completely convinced that the code is more readable this way.  Perhaps it is how my brain thinks, but the multiple calls to find, etc., could transparently be encapsulated in the single for loop that was there previously.